### DISCUSSION TITLE
Where are scripts stored, how to reload, etc.

### DISCUSSION THREAD

**Discussion by mikeerickson** [johnlindquist/kit#265] [Guide]
Hey John

Using this gem more and more each day.   I have a few scripts I would like to create and have a few questions (I have been trying to find this info in docs, etc. but not sure)

1.  Where are scripts scored (so that I can create scripts)
2.  Must I reload the application after creating / editing scripts
3. Must I do anything special to have custom scripts appear in list after pressing hotkey

**Comment by johnlindquist** [comment]
1. The scripts are in `~/.kenv/scripts`. "kenv" stands for "kit environment". 
> (When you "Edit" a script from the main menu, it should open the "~/.kenv" directory in your editor. If it doesn't, that's a bug.
2. No need to reload. They are literally "scripts", not apps. There's no compile step.
3. No, the Main Menu list reads from the `~/.kenv/scripts` directory. The app watches for changes in the directory and will regenerate the list automatically.

Hope that helps. Happy to answer any more questions!

**Reply by mikeerickson** [reply]
@johnlindquist Cool, that was easy!   What method should I use to display `alert` or `confirm`

FYI, I created  quick and dirty `Lorem Ipsum` generator

Ah, and one more question

- Does an interface exist to show a list of choices (in my case, I would like to prompt for `words`, `sentences`, `paragraphs`)

**Reply by johnlindquist** [reply]
Alert:
```js
await arg('Alert')
```

Confirm:
```js
await arg('Confirm', [{name: "No", value: false}, {name: "Yes", value: true}])
```

**Reply by johnlindquist** [reply]
Most basic list of choices:

```js
let value = await arg('Pick one', ["one", "two", "three"])
```

**Comment by mikeerickson** [comment]
@johnlindquist I am using `shared-selected-file.js` and am receiving an error about `getSelectedFile` not defined, which is coming from this line 14

```js
let filePath = await getSelectedFile()
```

**Reply by johnlindquist** [reply]
@mikeerickson a couple of scripts on the site are a bit outdated on the api/features. I'll go through them and clean them up after I get this new release out tomorrow.

**Comment by mikeerickson** [comment]
@johnlindquist OK, thank you kind sir, my script is all done and works like a charm!  If there is a user contribution section, I would be happy to submit (or if you would like to take a peek, let me know)
